.brigade/brigade.ts

```diff
@@ -285,6 +285,33 @@ const publishBrigadierDocsJob = (event: Event, version?: string) => {
 }
 jobs[publishBrigadierDocsJobName] = publishBrigadierDocsJob
 
+const testIntegrationJobName = "test-integration"
+const testIntegrationJob = (event: Event) => {
+  const job = new Job(testIntegrationJobName, "brigadecore/int-test-tools:v0.2.0", event)
+  job.primaryContainer.sourceMountPath = localPath
+  job.primaryContainer.workingDirectory = localPath
+  job.primaryContainer.environment = {
+    "DOCKER_REGISTRY": "localhost:5000",
+    "SKIP_DOCKER": "true",
+    "DOCKER_HOST": "localhost:2375",
+    "CGO_ENABLED": "0",
+    "BRIGADE_CI_PRIVATE_REPO_SSH_KEY": event.project.secrets.privateRepoSSHKey
+  }
+  job.primaryContainer.command = [ "sh" ]
+  job.primaryContainer.arguments = [
+    "-c",
+    // The sleep is a grace period after which we assume the DinD sidecar is
+    // probably up and running.
+    `sleep 20 && make hack-new-kind-cluster hack test-integration`
+  ]
+  job.sidecarContainers.docker = new Container(dindImg)
+  job.sidecarContainers.docker.privileged = true
+  job.sidecarContainers.docker.environment.DOCKER_TLS_CERTDIR=""
+  job.timeoutSeconds = 30 * 60
+  return job
+}
+jobs[testIntegrationJobName] = testIntegrationJob
+
```

Comment:
Overall strategy here is to do DinD just like we do for image builds. We launch the Docker daemon in a sidecar container. That means KinD runs in the sidecar.

I'm firing up an image registry (the same way we do with local/hack workflows) and making sure all KinD nodes use that. Using `kind load...`, for some reason, was not working. We kept getting `ImagePullBackoff`, but the other method worked with no problems.

Reply:
Approach sounds good to me.  fyi, when using 'kind load', image pull policy needs to be 'IfNotPresent' -- otherwise, kind always tries to pull images remotely, even if images have been loaded in.

Reply:
Oooooooh. That makes sense! I may revisit this in a follow-up then.